drivers/clock_control/clock_control_smartbond.c

```diff
@@ -231,8 +244,15 @@ static inline int smartbond_clock_control_off(const struct device *dev,
 		da1469x_clock_sys_xtal32m_init();
 		da1469x_clock_sys_xtal32m_enable();
 		break;
+	case SMARTBOND_CLK_USB:
+		/* Switch USB clock to HCLK to allow for resume */
+		CRG_TOP->CLK_CTRL_REG |= CRG_TOP_CLK_CTRL_REG_USB_CLK_SRC_Msk;
+		__fallthrough;
 	case SMARTBOND_CLK_PLL96M:
-		da1469x_clock_sys_pll_disable();
+		pll_requests &= ~(1 << (clk - SMARTBOND_CLK_PLL96M));
+		if (pll_requests == 0) {
```

Comment:
We should expected conflict with [that ](https://github.com/zephyrproject-rtos/zephyr/pull/73238/commits/b68d74e952be6505bbda59dfc5a6180d82ae2d68#diff-fc167ec9957914ba118ad236eb3862f8736edaf47c0c69a7422eaa4fbee28267R230) PR. I am fine with that approach but we should also call `da1469x_clock_check_device_div1_clock` before we close PLL and the clock is transparently switched back to XTAL32M. The mentioned HAL API also checks if USB is active and aborts. To avoid multiple paths I suggest that we open a HAL PR and remove code lines 143-151. That should be


```
    /* Exercise if USB is enabled and clocked by PLL */
    if (USB->USB_MCTRL_REG & USB_USB_MCTRL_REG_USBEN_Msk) {

        uint32_t clk_ctrl_reg = CRG_TOP->CLK_CTRL_REG;
        if (((clk_ctrl_reg & CRG_TOP_CLK_CTRL_REG_USB_CLK_SRC_Msk) == 0) &&
            ((clk_ctrl_reg & CRG_TOP_CLK_CTRL_REG_SYS_CLK_SEL_Msk) == 3)) {
                return true;
            }
    }
```

Reply:
What is more, I think the [on-off manager](https://docs.zephyrproject.org/latest/services/resource_management/index.html) would be a good fit here. If you think that it'd be a cheaper solution we could take advantage of it.

Reply:
@ioannis-karachalios it looks like *on-off manager* is present in Zephyr for over 4 years and is not used much except for few place in NRF related code. I think, I tried to incorporate this API in clock control for DA1469x at first but I abandoned this approach hence inclusion of onoff.h in our clock control that is not really used.

Reply:
Understood. If so, let's continue with the current approach.

Reply:
You need to re-base and resolve conflicts in the clock manager. I' have created a HAL PR that removes the USB check from `da1469x_clock_check_device_div1_clock` as a virtual USB clock is now introduced.

Reply:
thanks @ioannis-karachalios , I will rebase, currently I'm adding code that takes VBUS into consideration. I though that when there is no VBUS whole USB state machine can be stopped and additionally PD_SYS could be released so device can go to deep sleep if everything else is ready. When VBUS is present and host made suspend condition on USB lines, PLL can be stopped if not requested by non-USB code (current code in this PR already does it) but PD_SYS will be held. Third condition is when VBUS is present but application requested detach in this state I think driver can also release PD_SYS (but this I have to double check).

Reply:
As far as the third condition is concerned, that is when app. requests detach, It makes sense that the USB block should be disabled and PD_SYS should be released.

Reply:
As discussed before, `da1469x_clock_check_device_div1_clock ` has been updated and should no longer exercise the USB clock source.